src/Service/StripeManager.php

```diff
@@ -356,6 +354,8 @@ public function createSetupIntent(PaymentInterface $payment, $paymentMethod)
 
     public function attachPaymentMethodToCustomer(PaymentInterface $payment)
     {
+        $this->configure();
```

Comment:
@Atala this line is the actual fix. If `configure` method is not called an API key is not set on the Stripe client. 

All other changes is an attempt to limit such calls to the `StripeManager` class to avoid similar mistakes and avoid the need to "not forget" to call a `configure`method everywhere in the codebase